modules/dnn/test/test_gguf.cpp

```diff
@@ -0,0 +1,60 @@
+#include "test_precomp.hpp"
+#include "opencv2/dnn.hpp"
+#include <fstream>
+
+
+namespace opencv_test { namespace {
+// Helper function to mimic the _tf function.
+template<typename TString>
+static std::string _tf(TString filename, bool required = true)
+{
+    return findDataFile(std::string("dnn/gguf/") + filename, required);
+}
+
+
+
+class Test_GGUFImporter : public ::testing::Test
+{
+protected:
+    // You can do common initialization here if needed.
+};
+
+// This test creates a new GGUFImporter, parses a file using _tf to locate it,
+// then calls parse_attn_qkv() and checks that a Gemm layer has been added to the net.
+TEST_F(Test_GGUFImporter, readNetFromGGUF)
+{
+    // Locate the GGUF file; this should be in the directory dnn/gguf/ (adjust the filename as needed)
+    std::string ggufModelPath = _tf("mha.gguf", true);
+    std::string onnxModelPath = _tf("mha.onnx", true);
```

Comment:
It makes sense to save the model output as .pb file with ONNX runtime for example, but not relay on OpenCV implementation for ONNX. It's more reliable and easy to catch regressions.